Sources/TokamakCore/Modifiers/TextFieldStyle.swift

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Jed Fox on 06/30/2020.
+//
+
+public protocol TextFieldStyle {
+  // FIXME: Should properties be internal?
+  var type: String { get }
```

Comment:
These two properties seems very TokamakDOM specific.

Could you make this protocol empty, and extend it in TokamakDOM?

Here's what SwiftUIs `TextFieldStyle` looks like if you'd want to match that
```swift
public protocol TextFieldStyle {
  associatedtype _Body : SwiftUI.View
  func _body(configuration: SwiftUI.TextField<Self._Label>) -> Self._Body
  typealias _Label = SwiftUI._TextFieldStyleLabel
}

public struct _TextFieldStyleLabel : SwiftUI.View {
  public typealias Body = Swift.Never
}
```

Reply:
I’m not exactly sure how that works. How are you accessing these headers? (so I can look at them in the future)

Reply:
They're located inside of Xcode: `/Applications/Xcode-beta/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/System/Library/Frameworks/SwiftUI.framework/Modules/SwiftUI.swiftmodule/arm64.swiftinterface`

Super useful, especially when something is marked `@inlinable`

Reply:
As for how it works, all of these `XStyle` types seems to contain a `body` function with a configuration (which contains the label usually) passed in. This lets the user apply modifiers to the label to define the style of the component.

Reply:
aha I think I see what happens. The TextFieldStyle is wrapped around every TextField beneath it somehow by calling `someTextFieldStyle._body(textField)`, the output of which is swapped in for the actual `TextField`. Weirdly, `DefaultTextFieldStyle` is implemented differently, returning a custom view that wraps the actual body.

```swift
public struct RoundedBorderTextFieldStyle : SwiftUI.TextFieldStyle {
  public init()
  public func _body(configuration: SwiftUI.TextField<SwiftUI.RoundedBorderTextFieldStyle._Label>) -> some SwiftUI.View
  
  public typealias _Body = /* opaque return type */
}

public struct DefaultTextFieldStyle : SwiftUI.TextFieldStyle {
  public init()
  public func _body(configuration: SwiftUI.TextField<SwiftUI.DefaultTextFieldStyle._Label>) -> SwiftUI.DefaultTextFieldStyle._Body
  public struct _Body : SwiftUI.View {
    public var body: some SwiftUI.View { get }
    public typealias Body = /* opaque return type */
  }
}
```